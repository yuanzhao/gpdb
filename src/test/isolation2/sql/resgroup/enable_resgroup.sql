-- start_ignore
! rmdir /sys/fs/cgroup/cpu/gpdb;
! rmdir /sys/fs/cgroup/cpuacct/gpdb;
! mkdir /sys/fs/cgroup/cpu/gpdb;
! mkdir /sys/fs/cgroup/cpuacct/gpdb;
-- end_ignore

-- enable resource group and restart cluster.
-- start_ignore
! gpconfig -c gp_resource_manager -v group;
! gpstop -rai;
-- end_ignore

SHOW gp_resource_manager;

-- resource queue statistics should not crash
SELECT * FROM pg_resqueue_status;
SELECT * FROM gp_toolkit.gp_resqueue_status;
SELECT * FROM gp_toolkit.gp_resq_priority_backend;
