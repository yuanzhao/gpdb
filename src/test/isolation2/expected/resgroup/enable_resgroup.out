-- start_ignore
! rmdir /sys/fs/cgroup/cpu/gpdb;

! rmdir /sys/fs/cgroup/cpuacct/gpdb;

! mkdir /sys/fs/cgroup/cpu/gpdb;

! mkdir /sys/fs/cgroup/cpuacct/gpdb;

! chmod 755 /sys/fs/cgroup/cpu/gpdb;

! chmod 755 /sys/fs/cgroup/cpuacct/gpdb;

-- end_ignore

-- enable resource group and restart cluster.
-- start_ignore
! gpconfig -c gp_resource_manager -v group;
20170502:01:28:12:000367 gpconfig:sdw6:gpadmin-[WARNING]:-Managing queries with resource groups is an experimental feature. A work-in-progress version is enabled.
20170502:01:28:13:000367 gpconfig:sdw6:gpadmin-[INFO]:-completed successfully

! gpstop -rai;
-- end_ignore

SHOW gp_resource_manager;
gp_resource_manager
-------------------
group              
(1 row)

-- resource queue statistics should not crash
SELECT * FROM pg_resqueue_status;
rsqname|rsqcountlimit|rsqcountvalue|rsqcostlimit|rsqcostvalue|rsqwaiters|rsqholders
-------+-------------+-------------+------------+------------+----------+----------
(0 rows)
SELECT * FROM gp_toolkit.gp_resqueue_status;
queueid|rsqname|rsqcountlimit|rsqcountvalue|rsqcostlimit|rsqcostvalue|rsqmemorylimit|rsqmemoryvalue|rsqwaiters|rsqholders
-------+-------+-------------+-------------+------------+------------+--------------+--------------+----------+----------
(0 rows)
SELECT * FROM gp_toolkit.gp_resq_priority_backend;
rqpsession|rqpcommand|rqppriority|rqpweight
----------+----------+-----------+---------
(0 rows)
